src/data/reducers/grades.test.js

```diff
@@ -123,56 +117,180 @@ describe('grades reducer', () => {
       ...initialState,
       gradeFormat: formatTypeData,
     };
-    expect(grades(undefined, {
-      type: TOGGLE_GRADE_FORMAT,
-      formatType: formatTypeData,
-    })).toEqual(expected);
+    expect(grades(undefined, actions.toggleGradeFormat(formatTypeData)))
+      .toEqual(expected);
   });
 
   it('updates filter columns state success', () => {
-    const expectedHeadings = headingsData;
+    const expectedSelectedAssignmentType = 'selected assignment type';
     const expected = {
       ...initialState,
-      headings: expectedHeadings,
+      selectedAssignmentType: expectedSelectedAssignmentType,
+      headings: headingsData,
     };
-    expect(grades(undefined, {
-      type: FILTER_BY_ASSIGNMENT_TYPE,
-      headings: expectedHeadings,
-    })).toEqual(expected);
+    expect(grades(undefined, filterActions.update.assignmentType({
+      headings: headingsData,
+      filterType: expectedSelectedAssignmentType,
+    }))).toEqual(expected);
   });
 
+  it('updates fetch grades failure state', () => {
+    const expected = {
+      ...initialState,
+      errorFetching: true,
+      finishedFetching: true,
+    };
+    expect(grades(undefined, actions.fetching.error())).toEqual(expected);
+  });
+});
+
+describe('banner group', () => {
```

Comment:
These top-level groupings are confusing and not helpful.  Should be broken out under top-level if anything.